### PR TITLE
The Antenna module now improves comms in caves, loses req beacon

### PR DESCRIPTION
**Pull request by DeltaFire15**
## About The Pull Request
Alternate PR to #14594 

The antenna module loses its access to req beacons like in the referenced PR, but in exchange can reduce the interference of caves by one stage.
To do this, it first needs to be configured with a 30 second sync timer (which **does not** require you to stand in place), after which it will remain on indefinitely (barring unequipping it).
Reason for this channel is to avoid people "oceloting" an Antenna helmet to wear it when talking then take it back off. This makes it more of a commitment, and is a one-time thing as you very rarely have any reason to take your helmet off (never, in fact).

This means that in shallow caves, you will talk / hear like outside, and in deep caves you will hear like outside, but talk with some interference.
You also still have access to your location data, once your system is configured.

The implementation for receiving is a bit uhh because radios do not have a consistent place they are, and I wanted to include all radios that are anywhere on you in case you are carrying some special radio and not a headset, but should work.
Thanks to mandatory 515 I cannot test this PR because 515 would be detrimental to me, so I'll be using the CI to debug if I typod.
## Why It's Good For The Game
This is a much cooler solution for antenna modules instead of making them obsolete, while still requiring commitment due to a delay between equipping & enabling, and actually being to use it.

## Changelog
:cl:
balance: Antenna helmet modules have exchanged their ability to project a req drop beacon for improving caves communications (once configured)
/:cl:

### CODE REVIEW

**Comment by Sirsnek15** [comment]
"while still requiring commitment due to the one-time channel after equipping" One time channel? What? you mean like channeling as in the delay to put the antenna on? I think this is a great PR but theres no real 'commitment' there. Also what is "one time", you can take off and put on a module as many times as you want.

**Comment by DeltaFire15** [comment]
> "while still requiring commitment due to the one-time channel after equipping" One time channel? What? you mean like channeling as in the delay to put the antenna on? I think this is a great PR but theres no real 'commitment' there. Also what is "one time", you can take off and put on a module as many times as you want.

After equipping the helmet, the relay starts off, you have to use the action to activate it, which gives you a doafter (honestly could just be a timer though, it's mostly meant as a time delay), after which it will remain operational until you take the helmet off, which you should never do.

**Comment by novaepee** [comment]
> The entire point of antenna is to be a signal for req to send to though. This seems like something that would fit better on the radiopack.

Antenna was initially made to get coordinates on said marines #4641. Bravemole buffed it to let req send supplies to antenna #7023 .

**Comment by Sirsnek15** [comment]
> After equipping the helmet, the relay starts off, you have to use the action to activate it, which gives you a doafter (honestly could just be a timer though, it's mostly meant as a time delay), after which it will remain operational until you take the helmet off, which you should never do.
Thanks boss

**Comment by DeltaFire15** [comment]
Changed to be a timer instead (means you can keep moving and do not have an annoying doafter roundstart), however it is 30 seconds. (Anticipating people to try and juggle it for caves, thus that exists).

**Comment by efzapa** [comment]
isn't this an indirect MG nerf effectively 

unless you lug around a whole box of ammo, you're sorta fucked 

just buff the radio pack, (unless it already does the thing) or don't bother

**Comment by EvilMelonTheSecond** [comment]
Why would you fully replace an existing items function instead of making a new one? Why wouldn't you make this a new item instead of trying to tie this to another PR that might not get merged?

**Comment by SimplySmiley123** [comment]
> > The entire point of antenna is to be a signal for req to send to though. This seems like something that would fit better on the radiopack
> 
> this PR doesnt touch beacons (the entire role of beacons is to be a signal for req to send items to) nor does it touch radiopack (being the step above the beacon). It transforms the role of the antennae from being 'just a beacon but better in every way' to its own thing. +1

And effectively makes it useless.
I don't think cave pushing marines ever care about talking on comms, if they are coordinated enough where they would want to warn about I dunno a Queen, a King or whatever else, while pushing caves instead of doing the objectives, they will be sticking close together already.
While the less coordinated ungas won't use it anyway as they wouldn't care to say or listen to anyone anyway.

**Comment by Wisemonster** [comment]
> > this PR doesnt touch beacons (the entire role of beacons is to be a signal for req to send items to) nor does it touch radiopack (being the step above the beacon). It transforms the role of the antennae from being 'just a beacon but better in every way' to its own thing. +1
> 
> Beacons are ass to use and take up space and I don't even know where you get them without being an SL. Killing antenna just makes playing the game more of a pain as you have to either lug around boxes of ammo everywhere if you want to play the game, or hunt for someone with a beacon and bug them to place it down and wait with you so you can get your ammo and play the game.

Beacons are only 100 points at req, so if you really need one, you can order it from req. You could also ask an SL for their spare, considering each of them get 2 beacons. It being big is one of the trade-offs so that it isn't an easy-pick like the antenna currently is.

**Comment by Wisemonster** [comment]
I mean, people currently just use a spare helmet with an antenna module in their backpack if they want a different module to wear (such as full mimir set).

**Comment by SimplySmiley123** [comment]
> > > this PR doesnt touch beacons (the entire role of beacons is to be a signal for req to send items to) nor does it touch radiopack (being the step above the beacon). It transforms the role of the antennae from being 'just a beacon but better in every way' to its own thing. +1
> > 
> > 
> > Beacons are ass to use and take up space and I don't even know where you get them without being an SL. Killing antenna just makes playing the game more of a pain as you have to either lug around boxes of ammo everywhere if you want to play the game, or hunt for someone with a beacon and bug them to place it down and wait with you so you can get your ammo and play the game.
> 
> Beacons are only 100 points at req, so if you really need one, you can order it from req. You could also ask an SL for their spare, considering each of them get 2 beacons. It being big is one of the trade-offs so that it isn't an easy-pick like the antenna currently is.

Oh no I agree on that front.
It's just that this does leave the Antenna worthless to take.

**Comment by efzapa** [comment]
> >Beacons are only 100 points at req, so if you really need one, you can order it from req. You could also ask an SL for their spare, considering each of them get 2 beacons. It being big is one of the trade-offs so that it isn't an easy-pick like the antenna currently is.
> 
> We're needing to spend 100 req points and take up inventory space just to get the effect of *antenna* now? Not a chance in hell. This just makes the game more tedious for the sake of tedium.
> 
> Also, antenna is a popular module because all the helmet modules suck and barely make a difference so you may as well take antenna to make interacting with req not require you bug someone else to place their antenna (or post nerf, a fucking *beacon*).

coders try not to reqhell everything they dislike instead of improving it challenge (impossible)

**Comment by Technical-l** [comment]
> > > Beacons are only 100 points at req, so if you really need one, you can order it from req. You could also ask an SL for their spare, considering each of them get 2 beacons. It being big is one of the trade-offs so that it isn't an easy-pick like the antenna currently is.
> > 
> > 
> > We're needing to spend 100 req points and take up inventory space just to get the effect of _antenna_ now? Not a chance in hell. This just makes the game more tedious for the sake of tedium.
> > Also, antenna is a popular module because all the helmet modules suck and barely make a difference so you may as well take antenna to make interacting with req not require you bug someone else to place their antenna (or post nerf, a fucking _beacon_).
> 
> coders try not to reqhell everything they dislike instead of improving it challenge (impossible)

this is literally what this mfer is doing. Lumi is trying to completely kill antenna, and this guy is giving it a use instead of having it be destroyed

File: code/game/objects/items/radio/radio.dm [hunk 1]
```diff
@@ -292,10 +292,22 @@
...
+	SEND_SIGNAL(talking_movable, COMSIG_CAVE_INTERFERENCE_CHECK, inplace_interference)
+	radio_disruption = inplace_interference[1]
+
+	switch(radio_disruption)
+		if(CAVE_MINOR_INTERFERENCE)
+			signal.data["compression"] += rand(20, 40)
+		if(CAVE_FULL_INTERFERENCE)
 			return
-		signal.data["compression"] += rand(20, 40)
+		else
```
Comment (Lumipharon) [inline code/game/objects/items/radio/radio.dm:310]:
```suggestion
```

**Review by Lumipharon — APPROVED** [review]

File: code/game/objects/items/radio/radio.dm (hunk 1 above)
Comment (DeltaFire15) [reply code/game/objects/items/radio/radio.dm:310]: Personal style of mine to have an empty else in switch cases where one or more potential cases are not caught in the switch's ifs, can remove it though if disliked.

**Comment by Tommy-211** [comment]
https://images-ext-2.discordapp.net/external/Txv6vE3HQwi0POn6uSZG07TPMOu9eJj36D-jxDJTpZE/https/media.tenor.com/yBFx7v8GXSsAAAPo/joe-pesci-what-the-fuck.mp4

**Comment by Tommy-211** [comment]
Who the fuck pinged all the xeno mains to get them to upvote this

**Comment by Tommy-211** [comment]
mfw i check back on tgmc and im disappointed as always

**Pull request merged**